pkg/storage/tsdb/config.go

```diff
@@ -243,7 +243,7 @@ func (cfg *BucketStoreConfig) RegisterFlags(f *flag.FlagSet) {
 	cfg.BucketIndex.RegisterFlagsWithPrefix(f, "blocks-storage.bucket-store.bucket-index.")
 
 	f.StringVar(&cfg.SyncDir, "blocks-storage.bucket-store.sync-dir", "tsdb-sync", "Directory to store synchronized TSDB index headers.")
-	f.DurationVar(&cfg.SyncInterval, "blocks-storage.bucket-store.sync-interval", 5*time.Minute, "How frequently scan the bucket - or fetch the bucket index (if enabled) - to look for changes (new blocks shipped by ingesters and blocks removed by retention or compaction). 0 disables it.")
+	f.DurationVar(&cfg.SyncInterval, "blocks-storage.bucket-store.sync-interval", 5*time.Minute, "How frequently scan the bucket - or refresh the bucket index (if enabled) - to look for changes (new blocks shipped by ingesters and blocks deleted by retention or compaction).")
```

Comment:
**nit:** Adjust the phrasing slightly and use and em dash instead of an en dash.
```suggestion
	f.DurationVar(&cfg.SyncInterval, "blocks-storage.bucket-store.sync-interval", 5*time.Minute, "How frequently to scan the bucket -- or to refresh the bucket index (if enabled) -- in order to look for changes (new blocks shipped by ingesters and blocks deleted by retention or compaction).")
```

Reply:
While I agree with you, we currently don't use `--`  anywhere else :(

Reply:
I picked up the suggestion, changing `--` into `,`. Hope it works for you.